packages/core/theme-module/theme/pages/Category.vue

```diff
@@ -254,24 +267,47 @@ import {
   getProductVariants
 } from '<%= options.helpers %>';
 
+const defaultPagination = {
+  page: 1,
+  itemsPerPage: 20
+};
+
+const perPageOptions = [20, 40, 100];
+
 export default {
   transition: 'fade',
   setup(props, context) {
-    const { params } = context.root.$route;
+    const { params, query } = context.root.$route;
+    console.log(query);
     const lastSlug = Object.keys(params).reduce(
       (prev, curr) => params[curr] ? params[curr] : prev,
       params.slug_1
     );
 
     const { categories, search, loading } = useCategory('category-page');
-    const { products: categoryProducts, search: productsSearch, loading: productsLoading } = useProduct('category-products');
+    const { products: categoryProducts, totalProducts, search: productsSearch, loading: productsLoading } = useProduct('category-products');
+    const currentPage = ref(parseInt(query.page, 10) || defaultPagination.page);
+    const itemsPerPage = ref(parseInt(query.items, 10) || defaultPagination.itemsPerPage);
 
-    search({ slug: lastSlug });
+    search({ slug: lastSlug }).then(() => {
+      productsSearch({
+        catId: categories.value[0].id,
```

Comment:
Is there any option for empty categories array? If so then maybe: `(categories.value[0] || {}).id`